### ISSUE TITLE
Extend scope from commit to branch [enhancement]

### ISSUE THREAD

**Issue by GradedJestRisk** [commitizen/cz-cli#725]
You've done a great work. Do you think it can be extended to branch ?
When the user wants to create a branch, he would call a git bz

**Comment by dmwelch** [comment]
Do you mean something like [git-flow](https://github.com/nvie/gitflow)?

**Comment by GradedJestRisk** [comment]
Not quite. Look like git-flow enforce branch name by prefixing/appending fix type (feature/hotfix), but does not allow enforce using 
ticket number, application name and additional tags. It make sense that rule used for commit can be reused in branch.

Anyway, if gitflow indeed implement such feature (I don't knwo for sure, as I didn't install it), it would be interesting to tap in the same configuration file.

**Comment by GradedJestRisk** [comment]
Thanks for this extensive and detailed answer, I got your point.

**Comment by dmwelch** [comment]
No worries!
